packages/next-swc/crates/next-core/src/next_server/transforms.rs

```diff
@@ -60,6 +62,11 @@ pub async fn get_next_server_transforms_rules(
                 ActionsTransform::Server,
                 mdx_rs,
             ));
+
+            rules.push(get_next_react_server_components_transform_rule(
```

Comment:
might have to remove the `ClientDirectiveTransformer` added here: https://github.com/vercel/next.js/blob/96cc789f7df76fdac8533e68e9eade0498b0de55/packages/next-swc/crates/next-core/src/next_server/context.rs#L526-L528

maybe also add the server components transform there instead of here